presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -121,7 +147,7 @@ public boolean equals(Object obj)
                 Objects.equals(this.name, other.name) &&
                 Objects.equals(this.hiveColumnIndex, other.hiveColumnIndex) &&
                 Objects.equals(this.hiveType, other.hiveType) &&
-                Objects.equals(this.partitionKey, other.partitionKey);
+                Objects.equals(this.columnType, other.columnType);
```

Comment:
Validate `clientId` and `typeSignature` as well. I understand this is not directly related to your pull request. But this is a simple enough change.